Maths/Fraction.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @function fraction
+ * @description This function returns the fraction of a float type number.
+ * @param {number} num - a float type number is expected, but an integer will also work.
+ * @param {number} accuracy - the accuracy of the fraction, the default is 6.
+ * @return {Array} - an array containing the numerator and denominator of the fraction.
+ * @see https://en.wikipedia.org/wiki/Repeating_decimal and 
+ * @see https://en.wikipedia.org/wiki/Decimal#Decimal_fractions
+ * @example fraction(0.5) // [1, 2]
+ * @example fraction(0.3333333333333333) // [1, 3]
+ * @example fraction(0.25) // [1, 4]
+ * @example fraction(5.56) // [139, 25]
+ * @example fraction(0.33) // [33, 100]
+ */
+function fraction(number, accuracy = 6) {
+  let inp = typeof number === "number" ? number : false;
+  let acc = typeof accuracy === "number" ? accuracy : false;
+  if (inp !== false && acc !== false) {
+    if (Number.isInteger(inp)) {
+      return [inp, 1];
+    } else {
+      inp = inp.toString();
+      let len;
+      let reg = inp.match(/(\d+?)\1+$/);
```

Comment:
This string processing is awfully error prone (for example, what if JS decides to format your number using "nerd notation"?) and also hard to read. Replace it with proper number operations. For example, `Math.floor(number)` will be the integral part, and `number % 1` will be the fractional part.

Reply:
I got your point. But here is a but here. there are 2 types of rational numbers.
- Repetitive rational number. like: 1/3 = 0.333333...
- Non-Terminating rational number. like: 5/4 = 1.25

now your idea works for the second case easily, but for terminating numbers sometimes it doesn't work like that. like : 1.235235235... though it's a terminating rational number, but function returns 1235235235/1000000000 where it should return 1234/999. here regex decides that the number is terminating or not.

Reply:
There are periodic numbers, yes. But given a finite number, you can't know whether it is supposed to be periodic, and you shouldn't decide it arbitrarily.